actions/promote/tracker.js

```diff
@@ -0,0 +1,191 @@
+
+/**
+ * Initial version of tracker. More checks would be needed to cover few senarios
+ * 1. Checks if the activations are in progress.
+ * 2. If all activations are completed then manifest file is read to collect the failures.
+ * Few of changes to be done
+ * 1. Retry for few times if the activation is not present
+ * 2. Retry for few time if the manifest is not present.
+ */
+async function main(params) {
+    const ow = openwhisk();
+    logMemUsage();
+    let payload;
+    const {
+        adminPageUri, projectExcelPath, fgRootFolder, actDtls
+    } = params;
+    appConfig.setAppConfig(params);
+    const fgStatus = new FgStatus({ action: PROMOTE_ACTION, statusKey: fgRootFolder });
+    const filesSdk = await filesLib.init();
+    const batchManager = new BatchManager({ action: PROMOTE_ACTION, filesSdk });
+    try {
+        if (!fgRootFolder) {
+            payload = 'Required data is not available to proceed with FG Promote action.';
+            logger.error(payload);
+        } else if (!adminPageUri || !projectExcelPath) {
+            payload = 'Required data is not available to proceed with FG Promote action.';
+            await fgStatus.updateStatusToStateLib({
+                status: FgStatus.PROJECT_STATUS.FAILED,
+                statusMessage: payload
+            });
+            logger.error(payload);
+        } else {
+            urlInfo.setUrlInfo(adminPageUri);
+            payload = 'Getting status of all reference activation.';
+            fgStatus.updateStatusToStateLib({
+                status: FgStatus.PROJECT_STATUS.IN_PROGRESS,
+                statusMessage: payload
+            });
+            logger.info(`Activation to check ${JSON.stringify(actDtls)}`);
+
+            // Check to see all batches are complete
+            await checkBatches(fgRootFolder, actDtls, ow);
+            logger.info('Batch check comopleted proceed to mark complete');
+
+            // Collect status and mark as complete
+            await completePromote(projectExcelPath, actDtls, batchManager, fgStatus);
+
+            payload = 'Promoted floodgate tree successfully.';
+            logger.info(payload);
+        }
+    } catch (err) {
+        logger.error(err);
+        payload = err;
+        // In case of error log status with end time
+        try {
+            fgStatus.updateStatusToStateLib({
+                status: FgStatus.PROJECT_STATUS.COMPLETED_WITH_ERROR,
+                statusMessage: err.message,
+            });
+        } catch (err2) {
+            logger.info('Error while updatnig failed status');
+        }
+    }
+
+    return {
+        body: payload,
+    };
+}
+
+/**
+ * Wrapper that waits for all batches to complete
+ * @param {*} fgRootFolder Floodgatet folder path
+ * @param {*} actDtls Acttivation details of all batches
+ */
+async function checkBatches(fgRootFolder, actDtls, ow) {
+    let counter = 0;
+    while (await checkBatchesInProg(fgRootFolder, actDtls, ow) && counter < MAX_CHECKS) {
```

Comment:
@raga-adbe-gh As discussed, a scheduler might be better here:
Sample code:
https://git.corp.adobe.com/sukamat/playground-io/blob/main/app.config.yaml#L84-L100

Documentation:
https://developer.adobe.com/app-builder/docs/resources/cron-jobs/

cc: @Nethraasivakumar